From 84c5dadc32cc9db67398f0f274f4346e00ab9350 Mon Sep 17 00:00:00 2001
From: Danny Nold <dannynold@freescale.com>
Date: Wed, 4 May 2011 16:22:31 -0500
Subject: [PATCH 916/998] ENGR00141964 - EPDC fb: Ambient temperature setting handled incorrectly

- Added statement to ensure that when TEMP_USE_AMBIENT is selected as the
update temperature, the latest ambient temperature is used.  This code
was only handled incorrectly for the case where QUEUE_AND_MERGE update
scheme is used.

Signed-off-by: Danny Nold <dannynold@freescale.com>
---
 drivers/video/mxc/mxc_epdc_fb.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/video/mxc/mxc_epdc_fb.c b/drivers/video/mxc/mxc_epdc_fb.c
index 812af4a..f7ed434 100644
--- a/drivers/video/mxc/mxc_epdc_fb.c
+++ b/drivers/video/mxc/mxc_epdc_fb.c
@@ -2189,7 +2189,8 @@ static void epdc_submit_work_func(struct work_struct *work)
 		temp_index = mxc_epdc_fb_get_temp_index(fb_data,
 			upd_data_list->update_desc->upd_data.temp);
 		epdc_set_temp(temp_index);
-	}
+	} else
+		epdc_set_temp(fb_data->temp_index);
 	epdc_set_update_addr(upd_data_list->phys_addr
 				+ upd_data_list->update_desc->epdc_offs);
 	epdc_set_update_coord(adj_update_region.left, adj_update_region.top);
-- 
1.7.4.1

